### PR TITLE
add ZFS plugin

### PR DESCRIPTION
**Pull request by cornerot**
https://github.com/influxdb/telegraf/issues/317

### CODE REVIEW

File: plugins/zfs/zfs.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+	if len(kstatPath) == 0 {
+		kstatPath = "/proc/spl/kstat/zfs"
+	}
+
+	tags := getTags(kstatPath)
+
+	for _, metric := range kstatMetrics {
+		lines, err := common.ReadLines(kstatPath + "/" + metric)
+		if err != nil {
+			panic(err)
```
Comment (sparrc) [inline plugins/zfs/zfs.go]: We shouldn't panic all of Telegraf here, probably should just `return err`

File: plugins/zfs/zfs.go [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+}
+
+var sampleConfig = `
+  # ZFS kstat path
+  # If not specified, then default is:
+  # kstatPath = "/proc/spl/kstat/zfs"
+  #
+  # By default, telegraf gather all zfs stats
+  # If not specified, then default is:
+  # kstatMetrics = ["arcstats", "zfetchstats", "vdev_cache_stats"]
```
Comment (sparrc) [inline plugins/zfs/zfs.go]: just out of curiosity, why not these other stats?

```
/proc/spl/kstat/zfs# ll
total 0
dr-xr-xr-x 2 root root 0 Nov  3 17:43 ./
dr-xr-xr-x 3 root root 0 Nov  3 17:43 ../
-rw-r--r-- 1 root root 0 Nov  3 17:55 arcstats
-rw-r--r-- 1 root root 0 Nov  3 17:55 dbufs
-rw-r--r-- 1 root root 0 Nov  3 17:55 dmu_tx
-rw-r--r-- 1 root root 0 Nov  3 17:55 fm
-rw-r--r-- 1 root root 0 Nov  3 17:55 vdev_cache_stats
-rw-r--r-- 1 root root 0 Nov  3 17:55 xuio_stats
-rw-r--r-- 1 root root 0 Nov  3 17:55 zfetchstats
-rw-r--r-- 1 root root 0 Nov  3 17:55 zil
```

**Comment by cornerot** [comment]
I can add and the others, but these more demanded

**Comment by sparrc** [comment]
@cornerot don't worry about the other stats, but I need you to remove the panic before I can merge this. Could you also do me a favor and update the README and CHANGELOG?

thanks much!

**Comment by cornerot** [comment]
ok!

**Comment by cornerot** [comment]
CHANGELOG string:
```
- [#317](https://github.com/influxdb/telegraf/issues/317): ZFS plugin, thanks @cornerot!
```

**Pull request closed without merging**